### DISCUSSION TITLE
Improvement in documentation

### DISCUSSION THREAD

**Discussion by n-gram-hub** [godotengine/godot-proposals#6611] [General Discussions]
Hello, in a Godot forum, I posted a function where I use the `PackedVector2Array` _find_ method. The method searches for equal vectors and vectors can contain floats. I supposed this method was safe to use even because `PackedVector2Array` has no `is_equal_approx` method. But old users warned me not to use this method because maybe (maybe) the engine uses equals with floats, which can cause unpredictable results. 

So I'm here to ask you to specify in every corner of the docs what it is safe to use and what it is not: as a user, I can't say what the implementation of a method is like.

**Comment by AThousandShips** [comment]
I don't think this is needed, if nothing else is said one should assume equality is used rather than anything else, it says "searches for a value"

**Reply by n-gram-hub** [reply]
This is not a good approach, imho: I shouldn't even ask myself if I can use a method and if it is safe to use it. Moreover, in other areas of the docs, this kind of thing is specified, eg. where the docs talk about floats and comparisons.

**Reply by AThousandShips** [reply]
What made you assume it wasn't safe to use `find`?

**Reply by AThousandShips** [reply]
It's safe as long as you are aware of accuracy issues

**Reply by n-gram-hub** [reply]
I was quite sure it was safe before seasoned Godot (I'm not) users warned me it isn't because of an hypothetical issue with `==` at a lower level.

**Reply by AThousandShips** [reply]
Well they are wrong, it's safe, assuming you know the limitations of floating point accuracy

You need to make sure you aren't trying to find a vector that isn't there but making it the wrong value

**Reply by AThousandShips** [reply]
Well *your* code isn't safe because it doesn't do some checks, but that's not related to the floats, you need to check that the value was actually found

**Reply by n-gram-hub** [reply]
This is exactly what I think!

**Reply by AThousandShips** [reply]
To be clear: find is perfectly safe as long as you expect it to find the exact value you look for, taking float precision into account

**Reply by AThousandShips** [reply]
But you do need to format your code and add a check that the value was actually found

**Reply by n-gram-hub** [reply]
Well, the point coords come from a Godot method that returns the points in a curve. The array I search in comes from Godot that caches all the points of the curves.

**Reply by n-gram-hub** [reply]
I'm ok with the check, that was a forgetfulness due to haste

**Reply by AThousandShips** [reply]
Not knowing the discussion you had I'm not sure, but if you showed people this code their issue might have been that you did no check to actually see if `find` returned -1

**Reply by AThousandShips** [reply]
Their comments aren't on that it's unsafe in general, it's that you're using a curve with points in it and it's complicated

**Reply by AThousandShips** [reply]
And they're wrong about how it *should* use the approximate function, as above you expect it to return for exact values, or that it isn't clear that it uses equality for checking, the assumption it does, or even should, use approximate equality

**Comment by AThousandShips** [comment]
There *is* however a pitfall that the documentation of `Vector2` misses and that is `NaN`, which also shows up here for the `find` function, that *should* probably be documented

**Reply by n-gram-hub** [reply]
Thanks

**Reply by Calinou** [reply]
For reference, this was documented in https://github.com/godotengine/godot/pull/75614.
